api/v1beta1/githubendpoint_types.go

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: MIT
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// GitHubEndpointSpec defines the desired state of GitHubEndpoint
+type GitHubEndpointSpec struct {
+	Description   string `json:"description,omitempty"`
+	APIBaseURL    string `json:"apiBaseUrl,omitempty"`
+	UploadBaseURL string `json:"uploadBaseUrl,omitempty"`
+	BaseURL       string `json:"baseUrl,omitempty"`
+	CACertBundle  []byte `json:"caCertBundle,omitempty"` // TODO: This should be a secret reference
```

Comment:
@rafalgalaw have you already thought about that?

It's optional (from a garm POV) and only needed, if the garm server doesn't trust the CA.
At least in our case we do not need it. But for others it might be relevant ... 

I would prefer a secret ref as well

Reply:
as discussed, let's move this towards a secret ref and raise a new PR for this.

In this PR we will remove this field